### PR TITLE
x-pack/filebeat/input/http-endpoint: Add fallback content-type to incoming requests

### PR DESCRIPTION
**Pull request by bhapas**
## Proposed commit message

Few Webhooks like BitDefender do not send in `Content-Type` in their requests towards httpserver created in `http_endpoint` input. Hence the requests fail as the server expects a contentType.

Adding `application/json` as a fallback content-type for incoming requests if it is empty.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates https://github.com/elastic/integrations/issues/8696

### CODE REVIEW

**Comment by P1llus** [comment]
I think we might want to reuse the content type override functionality we already have, and find out why its not executing

**Comment by bhapas** [comment]
> I think we might want to reuse the content type override functionality we already have, and find out why its not executing

The `http.Handler` that is created to handle a specific endpoint pattern accepts `application/json` as a default. But the [`http.Request`](https://pkg.go.dev/net/http#Request) coming from external server does not have this set. If we have use the same config value as a fallback , then we can use it through the [apiValidator](https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/http_endpoint/validate.go#L32) struct in the handler and use it like

```diff
--- a/x-pack/filebeat/input/http_endpoint/handler.go
+++ b/x-pack/filebeat/input/http_endpoint/handler.go
@@ -54,7 +54,7 @@ type handler struct {
 func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+      // Few webhook servers don't send in the Content-Type , hence defaulting to application/json if not present
+        if r.Header.Get("Content-Type") == "" {
+               r.Header.Set("Content-Type", h.validator.contentType)
        }

```

WDYT?

**Comment by P1llus** [comment]
I just think that its weird its not already being set, its like part of the code is no longer being used after certain changes 🤔

File: x-pack/filebeat/input/http_endpoint/handler.go [hunk 1]
```diff
@@ -52,6 +52,11 @@ type handler struct {
 }
 
 func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Few webhook servers don't send in the Content-Type , hence defaulting to application/json if not present
+	if r.Header.Get("Content-Type") == "" {
```
Comment (andrewkroh) [inline x-pack/filebeat/input/http_endpoint/handler.go:56]: The only reason to modify the request is to make the validator pass. So why not disable content-type validation using configuration?

From the [docs](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-http_endpoint.html):

Screenshot 2024-01-24 at 13 34 14

**Review by andrewkroh — CHANGES_REQUESTED** [review]
I seems like this can be accomplished via configuration already.

**Comment by bhapas** [comment]
@andrewkroh Probably https://github.com/elastic/integrations/pull/8974 in favour of config changes

**Pull request closed without merging**